pubsub/rocketmq/rocketmq.go

```diff
@@ -70,60 +92,163 @@ func (r *rocketMQ) Init(metadata pubsub.Metadata) error {
 	opts := make([]mqc.Option, 0)
-	if r.metadata.ConsumerGroup != "" {
-		opts = append(opts, mqc.WithGroupName(r.metadata.ConsumerGroup))
+	if r.metadata.InstanceName != "" {
+		opts = append(opts, mqc.WithInstance(r.metadata.InstanceName))
+	}
+	if r.metadata.ConsumerGroupName != "" {
+		opts = append(opts, mqc.WithGroupName(r.metadata.ConsumerGroupName))
+	}
+	if r.metadata.NameServer != "" {
+		opts = append(opts, mqc.WithNameServer(strings.Split(r.metadata.NameServer, ",")))
 	}
 	if r.metadata.NameSpace != "" {
 		opts = append(opts, mqc.WithNamespace(r.metadata.NameSpace))
 	}
-	if r.metadata.Retries != 0 {
-		opts = append(opts, mqc.WithRetry(r.metadata.Retries))
-	}
 	if r.metadata.NameServerDomain != "" {
 		opts = append(opts, mqc.WithNameServerDomain(r.metadata.NameServerDomain))
 	}
-	if r.metadata.NameServer != "" {
-		opts = append(opts, mqc.WithNameServer(primitive.NamesrvAddr{r.metadata.NameServer}))
-	}
 	if r.metadata.AccessKey != "" && r.metadata.SecretKey != "" {
 		opts = append(opts, mqc.WithCredentials(primitive.Credentials{
-			AccessKey: r.metadata.AccessKey,
-			SecretKey: r.metadata.SecretKey,
+			AccessKey:     r.metadata.AccessKey,
+			SecretKey:     r.metadata.SecretKey,
+			SecurityToken: r.metadata.SecurityToken,
 		}))
 	}
-	return mq.NewPushConsumer(opts...)
+	if r.metadata.Retries > 0 {
+		opts = append(opts, mqc.WithRetry(r.metadata.Retries))
+	}
+	if r.metadata.ConsumerModel != "" {
+		if strings.EqualFold(r.metadata.ConsumerModel, "BroadCasting") {
+			opts = append(opts, mqc.WithConsumerModel(mqc.BroadCasting))
+		} else if strings.EqualFold(r.metadata.ConsumerModel, "Clustering") {
+			opts = append(opts, mqc.WithConsumerModel(mqc.Clustering))
+		} else {
+			r.metadata.ConsumerModel = "Clustering"
+			opts = append(opts, mqc.WithConsumerModel(mqc.Clustering))
+			r.logger.Warnf("%s Consumer Model[%s] is error, expected [BroadCasting, Clustering], "+
+				"we will use default model [Clustering]", r.name, r.metadata.ConsumerModel)
+		}
+	}
+	if r.metadata.FromWhere != "" {
+		if strings.EqualFold(r.metadata.FromWhere, "ConsumeFromLastOffset") {
+			opts = append(opts, mqc.WithConsumeFromWhere(mqc.ConsumeFromLastOffset))
+		} else if strings.EqualFold(r.metadata.FromWhere, "ConsumeFromFirstOffset") {
+			opts = append(opts, mqc.WithConsumeFromWhere(mqc.ConsumeFromFirstOffset))
+		} else if strings.EqualFold(r.metadata.FromWhere, "ConsumeFromTimestamp") {
+			opts = append(opts, mqc.WithConsumeFromWhere(mqc.ConsumeFromTimestamp))
+		} else {
+			r.metadata.FromWhere = "ConsumeFromLastOffset"
+			opts = append(opts, mqc.WithConsumeFromWhere(mqc.ConsumeFromLastOffset))
+			r.logger.Warnf("%s Consumer FromWhere[%s] is error, "+
+				"expected [ConsumeFromLastOffset, ConsumeFromFirstOffset, ConsumeFromTimestamp], "+
+				"we will use default value [ConsumeFromLastOffset]", r.name, r.metadata.FromWhere)
+		}
+	}
+	if r.metadata.ConsumeOrderly != "" {
+		if strings.EqualFold(r.metadata.ConsumeOrderly, "false") {
```

Comment:
Because the strings are all ASCII, you donìt need "equal fold" here. Instead, please use a regular `switch` statement which is easier to read.